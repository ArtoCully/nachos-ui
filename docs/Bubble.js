import React from 'react'
import { View } from 'react-native'

import Bubble from '../src/Bubble'
import H4 from '../src/typography/H4'

// WEBPACK (BubbleExample)
const BubbleExample = () => {
  const bubbleStyle = { margin: 10 }
  return (
    <View>
      <H4>Example:</H4>
      <View style={{ marginVertical: 15, flexDirection: 'row' }}>
        <Bubble style={bubbleStyle}>Hey, What’s Up?</Bubble>
        <Bubble
          style={bubbleStyle}
          arrowPosition='top'
          color='#ff9c00'
        >
          What’s cooking?
        </Bubble>
      </View>
    </View>
  )
}
// WEBPACK

const description = `
| Prop | Type | Default | Description |
|------|------|---------|-------------|
| \`arrowPosition\` | \`enum\` | \`left\` | Determines the position of the arrow. One of: \`'top', 'left', 'bottom', 'right'\` |
| \`color\` | \`string\` | \`'#2f8cff'\` | Background color of the Bubble compoment |
| \`style\` | \`object, array\` | \`void\` | Style of the Bubble compoment |
| \`theme\` | \`object\` | \`defaultTheme\` | An object describing a style of the component. See more in Theme section.  |
`

BubbleExample.styleguide = {
  ...BubbleExample.styleguide,
  title: 'Bubble',
  description,
}

export default BubbleExample
