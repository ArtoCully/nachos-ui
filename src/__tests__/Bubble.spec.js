import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Bubble from '../Bubble'

it('should render correctly', () => {
  const component = renderer.create(<Bubble>Hey, Whatâ€™s Up?</Bubble>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
