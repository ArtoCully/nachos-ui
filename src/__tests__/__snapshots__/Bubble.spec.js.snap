exports[`test should render correctly 1`] = `
<View
  style={undefined}>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 70,
          "paddingHorizontal": 25,
          "paddingVertical": 15,
          "position": "relative",
        },
        Object {
          "backgroundColor": "#2f8cff",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 14,
        }
      }>
      Hey, Whatâ€™s Up?
    </Text>
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
          },
        ]
      }>
      <View
        style={
          Object {
            "borderBottomColor": "transparent",
            "borderBottomWidth": 6,
            "borderRightColor": "#2f8cff",
            "borderRightWidth": 7,
            "borderTopColor": "transparent",
            "borderTopWidth": 6,
            "marginLeft": -7,
          }
        } />
    </View>
  </View>
</View>
`;
